File: crates/shared/src/import-rules/taxonomy_dependency_edge_vo.rs
// PURPOSE: DependencyEdge — Value Object representing a directed dependency edge between layers
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
pub struct DependencyEdge {
    pub source: String,
    pub target: String,
}

impl DependencyEdge {
    pub fn new(source: impl Into<String>, target: impl Into<String>) -> Self {
        Self {
            source: source.into(),
            target: target.into(),
        }
    }
}

File: packages/shared/src/taxonomy_dependency_edge_vo.ts
/** Taxonomy: Dependency edge value object — represents an import/dependency between files. */

export interface DependencyEdgeVO {
  readonly id: string;
  readonly source: string;
  readonly target: string;
  readonly type: "import" | "dependency";
}

export function createDependencyEdge(
  source: string,
  target: string,
  type: "import" | "dependency" = "import",
): DependencyEdgeVO {
  return {
    id: `${source}->${target}`,
    source,
    target,
    type,
  };
}
